lib/react_client.dart

```diff
@@ -734,8 +751,56 @@ class ReactJsComponentFactoryProxy extends ReactComponentFactoryProxy {
   }
 }
 
+class ReactDartFunctionComponentFactoryProxy extends ReactComponentFactoryProxy
+    with JsBackedMapsComponentFactoryBuilder {
+  /// The ReactJS function used as the type for all [ReactElement]s built by
+  /// this factory.
+  final Function reactFunction;
+
+  /// The name of this function.
+  final String displayName;
+
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  final ReactJsComponentFactory reactComponentFactory;
+
+  ReactDartFunctionComponentFactoryProxy(JsFunctionComponent reactFunction, [String displayName])
+      : this.reactFunction = reactFunction,
+        this.displayName = displayName,
+        this.reactComponentFactory = React.createFactory(reactFunction);
+
+  get type => reactFunction;
+}
+
+/// Creates a function component from the given [dartFunctionComponent] that can be used with React.
+JsFunctionComponent _wrapFunctionComponent(DartFunctionComponent dartFunctionComponent, {String displayName}) {
+  assert(() {
+    displayName ??= getProperty(dartFunctionComponent, 'name');
+    return true;
+  }());
+  jsFunctionComponent(JsMap jsProps, [JsMap _legacyContext]) =>
+      (dartFunctionComponent(jsProps != null ? JsBackedMap.backedBy(jsProps) : JsBackedMap()) ?? jsNull);
+  JsFunctionComponent interopFunction = allowInterop(jsFunctionComponent);
+  if (displayName != null) {
+    // This is a work-around to display the correct name in the React DevTools.
+    callMethod(getProperty(window, 'Object'), 'defineProperty', [
+      interopFunction,
+      'name',
```

Comment:
Nope, this is actually modifying the `name` property of the javascript function. because it is just a js function it doesnt have special properties like react component classes do, so its just the generic `name`.